### PR TITLE
Fix accidental no-shows on node restart

### PR DESCRIPTION
**Pull request by alexggh**
If approval was in progress we didn't actually restart it, so we end up in a situation where we distribute our assignment, but we don't distribute any approval.

### CODE REVIEW

**Comment by ordian** [comment]
tests

**Comment by alexggh** [comment]
Screenshot 2024-02-09 at 12 16 11

File: polkadot/node/core/approval-voting/src/lib.rs [hunk 1]
```diff
@@ -1396,16 +1407,54 @@ fn distribution_messages_for_activation(
...
+												.candidate_receipt()
+												.hash(),
+											indirect_cert,
+											assignment_tranche: assignment.tranche(),
+											relay_block_hash: block_hash,
+											session: block_entry.session(),
+											executor_params: executor_params.clone(),
+											candidate: candidate_entry.candidate_receipt().clone(),
+											backing_group: approval_entry.backing_group(),
+											distribute_assignment: false,
```
Comment (ordian) [inline polkadot/node/core/approval-voting/src/lib.rs:1455]: IIUC, alternatively, we could set this to true and not push into messages

**Review by ordian — APPROVED** [review]

File: polkadot/node/core/approval-voting/src/lib.rs [hunk 2]
```diff
@@ -1253,13 +1253,20 @@ async fn handle_actions<Context>(
 			Action::BecomeActive => {
 				*mode = Mode::Active;
 
-				let messages = distribution_messages_for_activation(
+				let (messages, next_actions) = distribution_messages_for_activation(
```
Comment (s0me0ne-unkn0wn) [inline polkadot/node/core/approval-voting/src/lib.rs:1256]:
```suggestion
				let (messages, mut next_actions) = distribution_messages_for_activation(
```

File: polkadot/node/core/approval-voting/src/lib.rs [hunk 3]
```diff
@@ -1253,13 +1253,20 @@ async fn handle_actions<Context>(
...
 					state,
 					delayed_approvals_timers,
-				)?;
+					session_info_provider,
+				)
+				.await?;
 
 				ctx.send_messages(messages.into_iter()).await;
+				let next_actions: Vec<Action> =
+					next_actions.into_iter().map(|v| v.clone()).chain(actions_iter).collect();
```
Comment (s0me0ne-unkn0wn) [inline polkadot/node/core/approval-voting/src/lib.rs:1267]:
```suggestion
				next_actions.extend(actions_iter);
```

It seemed like an unneeded complication unless I missed something.

**Review by s0me0ne-unkn0wn — APPROVED** [review]
Looks good! :+1:

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5339906

**Pull request merged**